virttest/qemu_vm.py

```diff
@@ -3454,7 +3454,11 @@ def create(self, name=None, params=None, root_dir=None,
                                                                    timeout)
                 except qemu_monitor.MonitorConnectError as detail:
                     LOG.error(detail)
+                    status = self.process.get_status()
```

Comment:
IMO, it's reasonable to raise a MonitorConnectError here, and the error msg is also explicit: Could not connect to monitor socket: [Errno 111] Connection refused, which means something wrong happened with the server, then we can try to find out what happened. Agreed with @YongxueHong 's comments, we cannot try to find out a server issue when hitting then monitor connection error here.

In qemu_img_lock_reject_boot, I am afraid we cannot simply depend on the VMCreateError exception, because we use aexpect.run_tail to run qemu-kvm in background, which may cost some time to quit. So we can put this check in the test case, we start vm2, then we wait some time or any exception till verify_alive is False, after that, we get the output and check the error msg. @luckyh @YongxueHong @yanan-fu @lkotek

Reply:
@zhencliu Thanks for comment! I see your point, to move solution of the problem from the `qemu_vm.py` directly to the test case is also an option and actually makes sense. It is true we hit this problem only with relation with `qemu_img_lock_reject_boot` and its specific use case with `luks`. I will give it a try :+1:

Reply:
> IMO, it's reasonable to raise a MonitorConnectError here, and the error msg is also explicit: Could not connect to monitor socket: [Errno 111] Connection refused, which means something wrong happened with the server, then we can try to find out what happened. Agreed with @YongxueHong 's comments, we cannot try to find out a server issue when hitting then monitor connection error here.
> 
> In qemu_img_lock_reject_boot, I am afraid we cannot simply depend on the VMCreateError exception, because we use aexpect.run_tail to run qemu-kvm in background, which may cost some time to quit. So we can put this check in the test case, we start vm2, then we wait some time or any exception till verify_alive is False, after that, we get the output and check the error msg. @luckyh @YongxueHong @yanan-fu @lkotek

Hi @zhencliu cc @lkotek 
I am afraid that we can not fix it in the workaround way from the tp_qemu since however, we wait time to get the exception, the exception is still the `virttest.qemu_monitor.MonitorConnectError: Could not connect to monitor 'qmpmonitor1'` which is not what we wanted. 
Because the error `e = virt_vm.VMCreateError(qemu_command, status, output)` still be skipped as you analyze.
So we might fix it in the avocado-vt framework.
That's just my analysis.
Anyway @lkotek you could have a try if it can be a workaround from `qemu_img_lock_reject_boot`
Thanks.

Reply:
Hi @lkotek and @zhencliu 
Oh, I forgot a workaround that we can try to check the `vm.process.get_output().strip()` after catching any exceptions  in the qemu_img_lock_reject_boot.
Thanks.
Like:
```
    try:
        vm2.create(params=vm2.params)
        vm2.verify_alive()
    except:
        output = output = self.process.get_output().strip()
        _verify_write_lock_err_msg(test, err.output, img_file)
```